### ISSUE TITLE
Refactor istgt_uctl_cmd_maxtime(UCTL_Ptr)

### ISSUE THREAD

**Issue by payes** [openebs/openebs#2128]
The following function istgt_uctl_cmd_maxtime(UCTL_Ptr):
https://github.com/openebs/istgt/blob/62d20a1ae272b4604359d47af06c1ff701f65033/src/istgt_lu_ctl.c#L2360-L2797
has a lot of duplicate code which needs to be optimized.
Addressing this will make our codebase more maintainable and improve [Better Code Hub](https://bettercodehub.com)'s **Write Simple Units of Code** guideline rating! 
Here's the gist of this guideline:
- **Definition** 📖 
Limit the number of branch points (if, for, while, etc.) per unit to 4.
- **Why**❓ 
Keeping the number of branch points low makes units easier to modify and test.
- **How** 🔧 
Split complex units with a high number of branch points into smaller and simpler ones.

You can find more info about this guideline in [Building Maintainable Software](http://shop.oreilly.com/product/0636920049159.do). 📖

**Comment by jayrajput** [comment]
I would like to work on it..Working as part of the hacktoberfest - Oct 20.

**Comment by sonasingh46** [comment]
Sure!

**Comment by jayrajput** [comment]
@payes and @sonasingh46 ...I am giving up on this one. I understand that we want to reduce the branch points and so I did tried to refactor the code into multiple functions. The end result did not looked good to me. The existing code looks better to me. And so the reason to not pursue breaking into multiple functions. The function has some lengthy sprintf which I also tried to break into multiple sprintf by looping, but it looked to me that single lengthy sprintf conveys the intent better than multiple sprintf.

There are three big things which the function does
- parse the arg and calculate iqn (iSCSI qualified name) and setzero (second optional arg)
- Process the optional second arg which basically set the total_time for first matching LU
- For each LU and IO type, set total_time and tdiff

Coding is part art and part science. Looks like the art here wins.
